### ISSUE TITLE
ChainingClassLoader - problem with custom kafka plugin

### ISSUE THREAD

**Issue by p53** [Graylog2/graylog2-server#8195]
We created our custom kafka plugin, but using some custom fields, we would like to add SASL there and upgrade kafka dependencies for it, but we have problems with that and reason is that you are using ChainingClassLoader, which uses graylog own classloader+adds classes from plugin jar. As graylog already contains kafka input with old libraries, these are before custom plugin own updated kafka-libraries in classloader, so plugin loads old instead of new one. 

https://github.com/Graylog2/graylog2-server/blob/20ab7325eb82ec09c08ece0433e183412f32a05d/graylog2-server/src/main/java/org/graylog2/shared/plugins/PluginLoader.java#L124

https://github.com/Graylog2/graylog2-server/blob/20ab7325eb82ec09c08ece0433e183412f32a05d/graylog2-server/src/main/java/org/graylog2/shared/plugins/PluginLoader.java#L55

Best solution would be probably make all inputs plugins

**Comment by bernd** [comment]
@p53 Yeah, moving inputs like the kafka ones into plugins would probably help. Out main problem is, that we also use the Kafka log implementation for our journal, so we cannot get rid of the Kafka dependency. We are planning to switch to a shaded Kafka dependency for that in the future to avoid conflicts with newer plugins.
